test/test_device_radix_sort.cu

```diff
@@ -946,36 +1093,47 @@ void TestBackend(
 template <bool IS_DESCENDING, typename KeyT, typename BeginOffsetIteratorT, typename EndOffsetIteratorT>
 void TestValueTypes(
     KeyT                 *h_keys,
-    int                  num_items,
+    size_t               num_items,
     int                  num_segments,
-    int                  *h_segment_offsets,
+    bool                 pre_sorted,
+    size_t               *h_segment_offsets,
     BeginOffsetIteratorT d_segment_begin_offsets,
     EndOffsetIteratorT   d_segment_end_offsets,
     int                  begin_bit,
     int                  end_bit)
 {
     // Initialize the solution
 
-    int *h_reference_ranks = NULL;
+    size_t *h_reference_ranks = NULL;
     KeyT *h_reference_keys = NULL;
-    InitializeSolution<IS_DESCENDING>(h_keys, num_items, num_segments, h_segment_offsets, begin_bit, end_bit, h_reference_ranks, h_reference_keys);
+    InitializeSolution<IS_DESCENDING>(h_keys, num_items, num_segments, pre_sorted, h_segment_offsets, begin_bit, end_bit, h_reference_ranks, h_reference_keys);
 
     // Test keys-only
     TestBackend<IS_DESCENDING, KeyT, NullType>          (h_keys, num_items, num_segments, d_segment_begin_offsets, d_segment_end_offsets, begin_bit, end_bit, h_reference_keys, h_reference_ranks);
 
     // Test with 8b value
     TestBackend<IS_DESCENDING, KeyT, unsigned char>     (h_keys, num_items, num_segments, d_segment_begin_offsets, d_segment_end_offsets, begin_bit, end_bit, h_reference_keys, h_reference_ranks);
 
-    // Test with 32b value
-    TestBackend<IS_DESCENDING, KeyT, unsigned int>      (h_keys, num_items, num_segments, d_segment_begin_offsets, d_segment_end_offsets, begin_bit, end_bit, h_reference_keys, h_reference_ranks);
+    if (!pre_sorted || sizeof(KeyT) <= 4)
```

Comment:
Why introduce this condition (and those that follow)?

Reply:
This is in order to avoid out-of-memory errors when running tests with a large number of items (e.g. 2 bln items).

I wonder if the condition should instead be based on the size of the GPU memory.

Reply:
These checks are now based on the amount of GPU memory.